Sources/MessagingInApp/Gist/Managers/QueueManager.swift

```diff
@@ -46,4 +51,16 @@ class QueueManager {
             completionHandler(.failure(error))
         }
     }
+
+    private func updatePollingInterval(headers: [AnyHashable: Any]) {
+        if let newPollingIntervalString = headers["x-gist-queue-polling-interval"] as? String,
+           let newPollingInterval = Double(newPollingIntervalString),
+           newPollingInterval != Gist.shared.messageQueueManager.interval {
+            DispatchQueue.main.async {
```

Comment:
why do we need `DispatchQueue.main` here? because the method `setup` is also using `DispatchQueue.main.asyncAfter`

Reply:
We're using that because of the polling timer being created in the `setup` call.